x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.ts

```diff
@@ -259,7 +260,7 @@ export const CredentialsLogic = kea<CredentialsLogicType>({
         const { http } = HttpLogic.values;
         await http.delete(`/api/app_search/credentials/${tokenName}`);
 
-        actions.onApiKeyDelete(tokenName);
+        actions.fetchCredentials();
```

Comment:
I changed this and removed `onApiKeyDelete` for the same reasons listed in this comment: https://github.com/elastic/kibana/pull/91565#discussion_r577148587

1. simpler code (xhr call over manually mutating the in-memory array)
2. handles pagination more correctly